### DISCUSSION TITLE
Interface materials for DG kernels

### DISCUSSION THREAD

**Discussion by ambehnam** [idaholab/moose#21819] [Q&A General] [answered]
Hello there,

I have created a DGkernel named VMNSDGKernel and want to apply it to the whole domain along with TotalLagrangianStressDivergence Kernel. The VMNSDGKernel needs to get some properties from a previously made interface material.  Is it possible to do so? I can't find any example that a DG kernel gets material properties from an interface material or any DGMaterial that inherit from InterfaceMaterial or Material.
In other words, for every element in the domain, Is it possible that the DG kernel get the face properties of element and neighbor from an interface material, without the need to breaking the mesh by blocks and without having interface boundary? If no, what are the alternatives?

When I tried to use my DG kernel with my previous interface material, I get the following error:
*** ERROR ***
The following error occurred in the object "tracaver_neighbor", of type "ComputeTractionAverJumpBubble".

MaterialBase::subdomainSetup in Material'tracaver_neighbor' needs to be implemented

Thanks a lot,
Amir

**Comment by lindsayad** [comment, marked as answer]
`InterfaceMaterials` are really only meant for use with `InterfaceKernels`. A regular `Material` should in most cases work fine with use of the `getMaterialProperty` and `getNeighborMaterialProperty` APIs. However, we do have an open issue for creation of an `InternalSideMaterial` at #14215 which is meant to solve the issues raised in #4335
